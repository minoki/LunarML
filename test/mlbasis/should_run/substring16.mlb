$(SML_LIB)/basis/basis.mlb
$(SML_LIB)/basis/text16.mlb
substring16.sml
