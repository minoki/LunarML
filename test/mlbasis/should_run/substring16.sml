infix ^^
val op ^^ = String16.^
val (a, b) = Substring16.splitl (fn c => c <> #"X") (Substring16.full "aaaXbbbbXccc")
val () = print (String16.toString (Substring16.string a ^^ "|" ^^ Substring16.string b) ^ "\n")
val (a, b) = Substring16.splitr (fn c => c <> #"X") (Substring16.full "aaaXbbbbXccc")
val () = print (String16.toString (Substring16.string a ^^ "|" ^^ Substring16.string b) ^ "\n")
val a = Substring16.triml 3 (Substring16.full "foobar")
val () = print (String16.toString (Substring16.string a) ^ "\n");
val a = Substring16.trimr 3 (Substring16.full "foobar")
val () = print (String16.toString (Substring16.string a) ^ "\n");
val s = Substring16.substring ("XfooXXbXarXX", 1, 10)
val () = print (String16.toString (String16.concatWith "," (List.map String16.str (Substring16.explode s))) ^ "\n")
val () = print (String16.toString (Substring16.translate (fn c => "[" ^^ String16.str (Char16.toUpper c) ^^ "]") s) ^ "\n")
val () = print (String16.toString (String16.concatWith "," (List.map Substring16.string (Substring16.tokens (fn x => x = #"X") s))) ^ "\n")
val () = print (String16.toString (String16.concatWith "," (List.map Substring16.string (Substring16.fields (fn x => x = #"X") s))) ^ "\n")
