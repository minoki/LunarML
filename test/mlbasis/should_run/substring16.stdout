aaa|XbbbbXccc
aaaXbbbbX|ccc
bar
foo
f,o,o,X,X,b,X,a,r,X
[F][O][O][X][X][B][X][A][R][X]
foo,b,ar
foo,,b,ar,
